### PR TITLE
Renamed type module to avoid clash with python builtin

### LINKED ISSUE #1624: hail.type overlaps a reserved function name
```
type(5)  # int. yay.
from hail import *
type(5)  # oops.
```

### PR DESCRIPTION
**Pull request by tpoterba**
Fixes #1624

### CODE REVIEW

**Comment by tpoterba** [comment]
Anyone want to review this?

**Comment by tpoterba** [comment]
er, wait

File: python/hail/docs/expr/index.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+type
```
Comment (cseed) [inline python/hail/docs/expr/index.rst]: This should say expr, right?

Comment (tpoterba) [inline python/hail/docs/expr/index.rst]: ah, yup. good catch

**Comment by cseed** [comment]
> er, wait

Anything else you're planning?  Looks go to me.

**Comment by tpoterba** [comment]
Nope, I think it's good now. The 'er, wait' was before I made docs fixes.

**Pull request merged**